nixos/modules/services/x11/display-managers/sddm.nix

```diff
@@ -14,7 +14,7 @@ let
   xserverWrapper = pkgs.writeScript "xserver-wrapper" ''
     #!/bin/sh
     ${concatMapStrings (n: "export ${n}=\"${getAttr n xEnv}\"\n") (attrNames xEnv)}
-    exec ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
+    exec systemd-cat ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
```

Comment:
@groxxda I may have misdescribed the situation at the end of #16096, sorry. Applications do indeed fail only on stop and start, not on restarts (as you've rightly mentioned above), but restarts perform okay. So the core problem was that `nixos-rebuild` (or, to be more specific, our activation script) did a bad job at restarting services with sockets, and also `systemd-journald` wasn't configured to restart instead of start and stop. This behaviour was fixed in #16096, as @aske points out.

The other question is if it's okay to break some applications when `systemd-journald` is stopped. I argue yes, because it's a critical service (something like `dbus`) and, want we or not, invasively persists everywhere nowadays. Either way, stopping it is _not_ recommended under any circumstances, just like `dbus`, so several application failures are to be expected.

Therefore I think that we can safely assume that `systemd-cat` performs well now and use it in various places -- if there are no leftover cases when `systemd-journald` is stopped in NixOS by some scripts.

BTW, @aske, you mentioned that `xserver` performs okay even if the pipe if broken. That's quite interesting! I think that may be because X server handles `SIGPIPE` correctly, which is good news.